distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkAction.java

```diff
@@ -0,0 +1,305 @@
+    private final String FORMATTED_ACTION_NOT_FOUND_ERROR_MSG = "This message does not contain a formatted DistributedNetworkAction";
+    private final String NOT_FORMATTED_PARAMS = "Params can't build a formatted DistributedNetworkAction";
+
+    public static final String DEFAULT_IGNORED = " ";
+
+    public static class Type {
+
+        /**
+         * @param action the int value to be checked
+         * @return true if the int matches an action, false if not
+         */
+
+        static private boolean isValid(int action) {
+
+            return action >= MIN_ACTION && action <= MAX_ACTION;
+
+        }
+
+
+        /**
+         * @param action the int value to be checked
+         * @return true if the given action uses the "PARAM" part of the message
+         */
+
+        static private boolean usesParam(int action) {
+
+            return isValid(action);
+
+        }
+
+
+        /**
+         * @param action the int value to be checked
+         * @return true if the given action uses the "EXTRA" part of message
+         */
+
+        static private boolean usesExtra(int action) {
+
+            return action == FIND_VALUE || action==ANSWER_FIND_VALUE;
+
+        }
+
+        /**
+         * @param action the int value to be checked
+         * @return true if the given action uses the "PAYLOAD" part of message
+         */
+
+        static private boolean usesPayload(int action) {
+
+            return action == FIND_VALUE || action==ANSWER_FIND_VALUE || action==STORE;
+
+        }
+
+        static private final int MIN_ACTION = 0;
+        static private final int MAX_ACTION = 9;
+        //<#>[ID] [k/N] [ACTION] [PARAM] [EXTRA] [PAYLOAD]
+
+        //<#>[ID] [1/1] PING [NODE_ID] [IGNORED] [IGNORED]
+        static final int PING = 0;
+        //<#>[ID] [1/1] <#>ANSWER_PING [NODE_ID] [IGNORED] [IGNORED]
+        static final int ANSWER_PING = 1;
+        //<#>[ID] [k/N] STORE [NODE_ID] [IGNORED] [VALUE]
+        static final int STORE = 2;
+        //<#>[ID] [k/N] ANSWER_STORE [NODE_ID] [IGNORED] [IGNORED]
+        static final int ANSWER_STORE = 3;
+        //<#>[ID] [k/N] FIND_NODE [NODE_ID] [IGNORED] [IGNORED]
+        static final int FIND_NODE = 4;
+        //<#>[ID] [k/N] ANSWER_FIND_NODE [NODE_ID] [IGNORED] [PHONENUMBER]
+        static final int ANSWER_FIND_NODE = 5;
+        //<#>[ID] [k/N] FIND_VALUE [NODE_ID] [YES_RESOURCE] [IGNORED]
+        static final int FIND_VALUE = 6;
+        //<#>[ID] [k/N] ANSWER_FIND_VALUE [NODE_ID] [YES_RESOURCE] [VALUE]
+        static final int ANSWER_FIND_VALUE = 7;
+        //<#>[ID] [k/N] INVITE [NODE_ID] [IGNORED] [IGNORED]
+        static final int INVITE = 8;
+        //<#>[ID] [k/N] ANSWER_INVITE [NODE_ID] [IGNORED] [IGNORED]
+        static final int ANSWER_INVITE = 9;
+    }
+
+    public DistributedNetworkAction setNodeId(int nodeId)
+    {
```

Comment:
Are you sure you need to set fields like this in this first 3 methods? Just setting them returning void should be enough